force-app/main/default/classes/CourseRegistrationHandler.cls

```diff
@@ -0,0 +1,135 @@
+                    }
+            else if(MyTriggers.hasChangedFields(fieldsToCheck, cr, triggerOldMap.get(cr.Id)) &&
+                    triggerOldMap.get(cr.Id).get('Status__c') == 'Venteliste')
+                    {
+                     courseId = cr.Course__c;
+                     reduceCourseWaitinglist(courseId);
+                    }
+            else if(MyTriggers.hasChangedFields(fieldsToCheck, cr, triggerOldMap.get(cr.Id)) &&
+                    cr.Status__c == 'Venteliste')
+                    {
+                        courseId = cr.Course__c;
+                        increaseCourseWaitinglist(courseId);
+                    }
+        }
+    }
+
+    public static void getCourseIdToReduce(List<CourseRegistration__c> registrations) {
+        Id courseId;
+        for (CourseRegistration__c cr : registrations) {
+            if(cr.Status__c == 'Påmeldt'){
+                courseId = cr.Course__c;
+                reduceCourseNumParticipants(courseId);
+            }
+            else if(cr.Status__c == 'Venteliste'){
+                courseId = cr.Course__c;
+                reduceCourseWaitinglist(courseId);
+            }
+        }
+    }
+
+    public static void reduceCourseNumParticipants(Id idToUpdate){
+        Course__c course = [SELECT Id, NumberOfParticipants__c FROM Course__c WHERE Id = :idToUpdate];
+
+        Decimal participantCount = course.NumberOfParticipants__c;
+        course.NumberOfParticipants__c = participantCount -1; 
+        update course;
+    }
+
+    public static void getCourseIdToIncrease(List<CourseRegistration__c> registrations) {
+        Id courseId;
+        for (CourseRegistration__c cr : registrations) {
+            if(cr.Status__c == 'Påmeldt'){
+                courseId = cr.Course__c;
+                increaseCourseNumParticipants(courseId);
+            }
+            else if(cr.Status__c == 'Venteliste'){
+                    courseId = cr.Course__c;
+                    increaseCourseWaitinglist(courseId);
+            }
+        }
+    }
+
+    public static void increaseCourseNumParticipants(Id idToUpdate){
+        Course__c course = [SELECT Id, NumberOfParticipants__c FROM Course__c WHERE Id = :idToUpdate];
+
+        Decimal participantCount = course.NumberOfParticipants__c;
+        course.NumberOfParticipants__c = participantCount +1; 
+        update course;
+    }
+
+    public static void reduceCourseWaitinglist(Id idToUpdate){
+        Course__c course = [SELECT Id, Waitinglist__c FROM Course__c WHERE Id = :idToUpdate];
+
+        Decimal waitinglistCount = course.Waitinglist__c;
+        course.Waitinglist__c = waitinglistCount -1; 
+        update course;
+    }
+
+    public static void increaseCourseWaitinglist(Id idToUpdate){
+        Course__c course = [SELECT Id, Waitinglist__c FROM Course__c WHERE Id = :idToUpdate];
+
+        Decimal waitinglistCount = course.Waitinglist__c;
+        course.Waitinglist__c = waitinglistCount +1; 
+        update course;
+    }
```

Comment:
Her har du MANGE SOQL-queries inni en for-loop.

Det enkleste vil være å bytte fra after insert/update/delete/undelete, å heller bruke before. Da trenger du ikke kjøre `update` på noen records, da endrer du bare verdien på rekorden så vil verdien automatisk lagres.